### PR TITLE
Fix #132 module_file() freeze

### LINKED ISSUE #132: 100% cpu
# Problem summary

When editing the following file, i get 100% cpu usage (for people like me who have some python plugins that use synchronous functions, neovim freeze forever)

```python
#!/usr/bin/env python3

"""
./A.B
"""

"""
import C
"""
```

## Expected
normal completion, no 100% cpu usage

## Environment Information

 * OS: archlinux
 * Neovim version: NVIM v0.2.1-870-g1551f7132

## Provide a minimal init.vim with less than 50 lines (required)

```vim
let s:config_dir = expand('~/.config/nvim/')
let s:bundle_dir = s:config_dir . '/bundle/'

call plug#begin(s:bundle_dir)

Plug 'zchee/deoplete-jedi'
Plug 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }

call plug#end()

let g:deoplete#enable_at_startup = 1
```

## Steps to reproduce the issue after starting Neovim (required)

 /B\<CR>s

Without the import part, everything works fine (i don't understand anything that is happening...)

### PR DESCRIPTION
**Pull request by Shougo**
@zchee Ping.

### CODE REVIEW

**Comment by Shougo** [comment]
@RageYL Please test it.

File: rplugin/python3/deoplete/sources/deoplete_jedi/utils.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def module_file(dirname, suffix, base):
         if os.path.isfile(p):
             found = p
             break
-        if os.path.basename(search) == base:
+        if os.path.basename(search) == base or search == dirname:
```
Comment (Shougo) [inline rplugin/python3/deoplete/sources/deoplete_jedi/utils.py]: I don't know the code.
`os.path.basename(search) == base` is really needed?

**Comment by RageYL** [comment]
I can confirm the issue has been fixed ! Thanks

**Comment by Shougo** [comment]
Rebased.

@zchee Ping

**Comment by zchee** [comment]
@Shougo I’ll check it tomorrow, thanks.

**Comment by Shougo** [comment]
@zchee Ping.

**Comment by blueyed** [comment]
What issue is this about?

@Shougo 
Can you link to or provide a test case?

**Comment by blueyed** [comment]
Ah, https://github.com/zchee/deoplete-jedi/issues/132.

Since it is confirmed by the reporter I think we should merge it.

**Comment by Shougo** [comment]
OK.  I will merge it later when flake8 error is fixed.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**